lib/Models/ArcGisMapServerCatalogItem.js

```diff
@@ -195,6 +203,30 @@ function getJson(item, uri) {
     return loadJson(proxyCatalogItemUrl(item, uri.addQuery('f', 'json').toString(), '1d'));
 }
 
+function getToken(item) {
+    const options = {
+        url: item.tokenUrl,
+        method: 'POST',
+        headers: {'Content-Type': 'application/json'},
+        data: JSON.stringify({'url': item.url})
+    };
+
+    return loadWithXhr(options).then(function (result) {
+        const token = JSON.parse(result);
+
+        if (defined(token.token)) {
+            item._lastToken = token.token;
+            return token.token;
+        }
+
+        item._lastToken = undefined;
+        throw new RuntimeError('The token server responded with an invalid token.');
+    }).otherwise(() => {
+        item._lastToken = undefined;
+        throw new RuntimeError('Unable to request a token from the token server.');
```

Comment:
Probably better to throw `TerriaError` here (and above).

Reply:
Thanks.

5bcbe3e2fe6c8272d7239ccd0169b41a30f4e581